pkg/kb_agent/handlers/exec_handler.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright (C) 2022-2024 ApeCloud Co., Ltd
+
+This file is part of KubeBlocks project
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package handlers
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/apecloud/kubeblocks/pkg/kb_agent/util"
+)
+
+type ExecHandler struct {
+	Logger logr.Logger
+
+	Executor util.Executor
+}
+
+var _ Handler = &ExecHandler{}
+
+func NewExecHandler(properties map[string]string) (*ExecHandler, error) {
+	logger := ctrl.Log.WithName("EXEC handler")
+
+	h := &ExecHandler{
+		Logger:   logger,
+		Executor: &util.ExecutorImpl{},
+	}
+
+	return h, nil
+}
+
+func (h *ExecHandler) Do(ctx context.Context, setting util.Handlers, args map[string]any) (map[string]any, error) {
+	if len(setting.Command) == 0 {
+		h.Logger.Info("action command is empty!")
+		return nil, nil
+	}
+	envs := util.GetAllEnvs(args)
+	h.Logger.Info("execute action", "commands", setting.Command, "envs", envs)
+	output, err := h.Executor.ExecCommand(ctx, setting.Command, envs)
```

Comment:
Should we add timeout protection?

Reply:
i think it is better to add timeout protection for the following reasons:
- Timeout is important to prevent commands from hanging or getting stuck, particularly for actions that run periodically. 
- Timeout lets users know whether the action executed as expected. If an action fails to complete within the specified timeframe, a fail event can be emitted.